src/Plugins/BotSharp.Plugin.EmailHandler/Functions/HandleEmailReaderFn.cs

```diff
@@ -0,0 +1,205 @@
+using BotSharp.Abstraction.Files;
+using BotSharp.Abstraction.Messaging.Models.RichContent.Template;
+using BotSharp.Abstraction.MLTasks;
+using BotSharp.Core.Infrastructures;
+using BotSharp.Plugin.EmailHandler.Models;
+using BotSharp.Plugin.EmailHandler.Providers;
+using MailKit;
+using MailKit.Net.Imap;
+using MailKit.Search;
+using MailKit.Security;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using MimeKit;
+
+namespace BotSharp.Plugin.EmailReader.Functions;
+
+public class HandleEmailReaderFn : IFunctionCallback
+{
+    public string Name => "handle_email_reader";
+    public readonly static string PROMPT_SUMMARY = "Provide a text summary of the following content.";
+    public readonly static string RICH_CONTENT_SUMMARIZE = "is_email_summarize: true. messageId";
+    public readonly static string RICH_CONTENT_READ_EMAIL = "Read the email by messageId";
+    public readonly static string RICH_CONTENT_MARK_READ = "mark_as_read: true. messageId";
+    public string Indication => "Handling email read";
+    private readonly IServiceProvider _services;
+    private readonly ILogger<HandleEmailReaderFn> _logger;
+    private readonly IHttpContextAccessor _context;
+    private readonly BotSharpOptions _options;
+    private readonly EmailReaderSettings _emailSettings;
+    private readonly IConversationStateService _state;
+    private readonly IEmailReader _emailProvider;
+
+    public HandleEmailReaderFn(IServiceProvider services,
+                ILogger<HandleEmailReaderFn> logger,
+                IHttpContextAccessor context,
+                BotSharpOptions options,
+                EmailReaderSettings emailPluginSettings,
+                IConversationStateService state,
+                IEmailReader emailProvider)
+    {
+        _services = services;
+        _logger = logger;
+        _context = context;
+        _options = options;
+        _emailSettings = emailPluginSettings;
+        _state = state;
+        _emailProvider = emailProvider;
+    }
+    public async Task<bool> Execute(RoleDialogModel message)
+    {
+        var args = JsonSerializer.Deserialize<LlmContextReader>(message.FunctionArgs, _options.JsonSerializerOptions);
+        var isMarkRead = args?.IsMarkRead ?? false;
+        var isSummarize = args?.IsSummarize ?? false;
+        var messageId = args?.MessageId;
+        try
+        {
+            if (!string.IsNullOrEmpty(messageId))
+            {
+                if (isMarkRead)
+                {
+                    await _emailProvider.MarkEmailAsReadById(messageId);
+                    message.Content = $"The email message has been marked as read.";
+                    return true;
+                }
+                var emailMessage = await _emailProvider.GetEmailById(messageId);
+                if (isSummarize)
+                {
+                    var prompt = $"{PROMPT_SUMMARY} The content was sent by {emailMessage.From.ToString()}. Details: {emailMessage.TextBody}";
```

Comment:
The prompt should be placed under `Utility Agent` templates folder. This will allow user to updated the prompt without updating code.